modules/tensor_mechanics/src/base/TensorMechanicsApp.C

```diff
@@ -342,6 +343,8 @@ TensorMechanicsApp::associateSyntax(Syntax & syntax, ActionFactory & action_fact
   syntax.registerActionSyntax("EmptyAction", "BCs/Pressure");
   syntax.registerActionSyntax("PressureAction", "BCs/Pressure/*");
 
+  syntax.registerActionSyntax("GeneralizedPlaneStrainAction", "GeneralizedPlaneStrain/*");
```

Comment:
Yes, I believe @dschwen is already doing that in phase-field. We don't have to do that in this PR but we should be moving that direction. If we are adding new syntax, I'd suggest that we put it in the right place rather than moving it later after it starts appearing in input files.